src/Core/Domain/Profile/Exception/ProfileConstraintException.php

```diff
@@ -0,0 +1,43 @@
+<?php
+/**
+ * 2007-2019 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\Profile\Exception;
+
+/**
+ * Is thrown when some constraint is violated in Profile subdomain
+ */
+class ProfileConstraintException extends ProfileException
+{
+    /**
+     * Profile name max length as defined in the ObjectModel
+     */
+    const NAME_MAX_LENGTH = 32;
```

Comment:
I think it would make more sense to put this in AddProfileCommand

Reply:
Yes, I thought about it but since we need to check the length in `EditProfileCommand` too, I thought the common place would have been the Exception linked to that check.
So should I use `AddProfileCommand::MAX_LENGTH` in `EditProfileCommand` or should I add a second constante in `EditProfileCommand`?

Reply:
For this kind of constants, we have also been using "Settings" class
See https://github.com/PrestaShop/PrestaShop/blob/develop/src/Core/Domain/Category/SeoSettings.php

Reply:
Agreed! This kind of domain specification should not be stored in the exception